### DISCUSSION TITLE
Neorv32 complex setups

### DISCUSSION THREAD

**Discussion by zipotron** [stnolting/neorv32#220] [Show and tell]
Hello. I have created a repo for implement the SDRAM driver for ULX3S and other complex setups here:
https://github.com/zipotron/neorv32-complex-setups
Please, feel free to give opinions about how is being done.
Thanks!

**Comment by stnolting** [comment]
Looks good so far!
So this is a setup with SDRAM, right? Some more details in the README would be nice, maybe including a memory map.
If you like you can add that repository to the list of "external" setups in [`setups`](https://github.com/stnolting/neorv32/tree/master/setups).

**Reply by zipotron** [reply]
Hi! I just added in the list of "external" setups, is part of the PR #213!

**Comment by zipotron** [comment]
Thanks! You are right! I will, just need time, also, before, I need to finish the TOP, I could set the PLL, but the wishbone bus is not wired yet...

**Comment by umarcor** [comment]
@zipotron instead of copying a BoardTop into the submodule, it should be possible to override BOARD_SRC (https://github.com/stnolting/neorv32/blob/629454017918da9953bb54492be1436a82b80e8d/setups/osflow/Makefile#L19). Maybe it needs an adjustment similar to stnolting/neorv32#214.

**Reply by zipotron** [reply]
This is a good idea... I will investigate! Thanks

**Reply by zipotron** [reply]
I tried, but I think my Makefile knowledge in not enough deep for do it

**Reply by zipotron** [reply]
I will give a second try in the incoming weeks...
